### PR TITLE
temple-glow-format - Add support/tools for mapping temple light sources

### PR DESCRIPTION
**Pull request by raffclar**
This PR adds support for the "glw" file format. This format directs the position & properties of the light sources (glows) within the temple interior.

Most of the known fields are from fuzzing the format and inspecting vanilla ingame changes so churn with the structs are to be expected until we know for certain which bytes account for what.

### CODE REVIEW

File: src/ECS/Archetypes/GlowArchetype.cpp
Comment (bwrsandman) [reply src/ECS/Archetypes/GlowArchetype.cpp:26]: This should not be a suggested fix since those are included in the header....

File: src/Renderer.cpp
Comment (bwrsandman) [inline src/Renderer.cpp:752]: This looks like it was negated accidentally

File: src/Resources/Loaders.h
Comment (bwrsandman) [inline src/Resources/Loaders.h:15]: Expensive header and not needed
```suggestion
```

**Review by bwrsandman — CHANGES_REQUESTED** [review]

**Comment by raffclar** [comment]
I'll do a final rebase before merge

**Comment by bwrsandman** [comment]
Still the issue with `desc.drawTestModel`

**Comment by raffclar** [comment]
> Still the issue with `desc.drawTestModel`

Sorted and rebased. I think we're good to go once CI passes

**Comment by bwrsandman** [comment]
Is there a depth test missing?

image

File: src/Renderer.cpp
Comment (bwrsandman) [inline src/Renderer.cpp:720]:
```suggestion
					    bgfx::setState(0 | BGFX_STATE_DEPTH_TEST_GREATER | BGFX_STATE_WRITE_RGB | BGFX_STATE_WRITE_A |
					                   BGFX_STATE_BLEND_FUNC(BGFX_STATE_BLEND_SRC_ALPHA, BGFX_STATE_BLEND_ONE) |
					                   BGFX_STATE_BLEND_EQUATION(BGFX_STATE_BLEND_EQUATION_ADD));
```

File: src/Renderer.cpp
Comment (bwrsandman) [inline src/Renderer.cpp:751]: If this isn't used, then it should be deleted
```suggestion
```

File: src/Renderer.h
Comment (bwrsandman) [inline src/Renderer.h:79]:
```suggestion
```

File: src/Game.cpp
Comment (bwrsandman) [inline src/Game.cpp:854]:
```suggestion
```

File: src/Game.h
Comment (bwrsandman) [inline src/Game.h:131]:
```suggestion
```

File: src/Profiler.h
Comment (bwrsandman) [inline src/Profiler.h:44]:
```suggestion
```

File: src/Profiler.h
Comment (bwrsandman) [inline src/Profiler.h:52]:
```suggestion
```

**Review by bwrsandman — CHANGES_REQUESTED** [review]
- [ ] Fix the inverse depth
- [ ] Simplify `BGFX_STATE_BLEND_FUNC`
- [ ] Remove `drawGlowEffects` config and profiler category

**Review by bwrsandman — APPROVED** [review]

**Pull request merged**
